lib/similar.js

```diff
@@ -1,28 +1,93 @@
 'use strict';
 
 const request = require('./utils/request');
-const cheerio = require('cheerio');
+const R = require('ramda');
+const url = require('url');
+const queryString = require('querystring');
+const matchScriptData = require('./utils/matchScriptData');
+const PLAYSTORE_URL = 'https://play.google.com/store/apps/details';
 
-function similar (getParseList, opts) {
+function similar (opts) {
   return new Promise(function (resolve, reject) {
     if (!opts || !opts.appId) {
       throw Error('appId missing');
     }
 
+    opts.appId = encodeURIComponent(opts.appId);
     opts.lang = opts.lang || 'en';
-    const appId = encodeURIComponent(opts.appId);
+    opts.country = opts.country || 'us';
+
+    const qs = queryString.stringify({
+      id: opts.appId,
+      hl: opts.lang,
+      gl: opts.country
+    });
+    const reqUrl = `${PLAYSTORE_URL}?${qs}`;
 
     const options = Object.assign({
-      url: `https://play.google.com/store/apps/similar?id=${appId}&hl=${opts.lang}`,
+      url: reqUrl,
       followAllRedirects: true
     }, opts.requestOptions);
 
     request(options, opts.throttle)
-      .then(cheerio.load)
-      .then(getParseList(opts))
+      .then(matchScriptData)
+      .then(data => {
+        const link = R.path(['ds:6', 1, 1, 0, 0, 3, 4, 2], data);
+        if (link === undefined) {
+          throw Error('not found similar link');
+        }
+        return link;
+      })
+      .then(link => {
+        return request(Object.assign({
+          url: `https://play.google.com${link}&gl=${opts.country}&hl=${opts.lang}`,
+          followAllRedirects: true
+        }, opts.requestOptions));
+      })
+      .then(matchScriptData)
+      .then(extractFields)
+      .then(data =>
+        data.map(e => {
+          const q = url.parse(e.developerId, { parseQueryString: true });
+          e['developerId'] = q.query['id'];
+          e['url'] = new url.URL(e['url'], 'https://play.google.com').toString();
+          if (e['priceText'] === undefined) {
```

Comment:
Can you handle all this processing declaratively as it's done in the app.js module? If fields require extra processing use mapping functions like this: 

https://github.com/facundoolano/google-play-scraper/blob/044f320cee8347481fd077543fe3d147a653c62f/lib/app.js#L79-L82